storage_service/administration/forms.py

```diff
@@ -222,10 +222,30 @@ class UserChangeForm(auth.forms.UserChangeForm):
     """ Modifys an existing user.  Inherits from django's UserChangeForm. """
 
     def __init__(self, *args, **kwargs):
+        current_user = kwargs.pop("current_user", None)
+        user_being_edited = kwargs["instance"]
+        self.superusers = auth.get_user_model().objects.filter(is_superuser=True)
         super(UserChangeForm, self).__init__(*args, **kwargs)
         self.fields["is_superuser"].label = _("Administrator?")
+        if not (current_user and current_user.is_superuser):
+            # If current user is not super, do not permit editing of that.
+            del self.fields["is_superuser"]
+        elif self.superusers.count() == 1 and not (current_user != user_being_edited):
+            # Provide some indication that this is undesirable."""
+            self.fields["is_superuser"].widget.attrs["readonly"] = True
         del self.fields["password"]
 
+    def clean(self):
+        """Validate the form to protect against potential user errors."""
+        if self.superusers.count() > 1:
+            return self.cleaned_data
+        try:
+            # Protect field from being reverted if only one superuser.
+            self.cleaned_data["is_superuser"] = True
```

Comment:
What about raising a [ValidationError](https://docs.djangoproject.com/en/1.8/ref/forms/validation/#raising-validationerror) here instead? That should give the user some feedback as to what's happening.

I think if you go that route it actually removes the need for the readonly widget I suggested before (sorry!).

Reply:
I think this is a good suggestion, but have left it as is for now. The validation error looks a bit messy as it will complain password fields need changing too, and I think possibly protects the form from being changed in other ways. 

**NB.** for transparency as I merge, I put in one additional check here after the try to make sure this only impacts the user being edited if they are already a super user as I spotted all users would otherwise get made super if editing.